### ISSUE TITLE
Discuss whether DeclarationKind.RecordStruct should be shipped or deleted

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#52799]
See https://github.com/dotnet/roslyn/issues/48136#issuecomment-823832424 for context.

cc @jcouv @sharwell @CyrusNajmabadi

**Comment by sharwell** [comment]
It's my preference to delete `RecordStruct`, and mark `RecordClass` as:

```
[Obsolete($"This value is not used. Use {nameof(Class)} instead.")]
[EditorBrowsable(EditorBrowsableState.Never)]
```

**Comment by CyrusNajmabadi** [comment]
I'm fine with that.

**Comment by jcouv** [comment]
If any changes happen relative to records in the next week or two, please make the base the change off of the `features/record-structs` branch to avoid creating more conflicts.

**Comment by sharwell** [comment]
@jcouv This is technically a change, but based on the PR it seems unlikely to conflict

**Comment by jcouv** [comment]
I'm saying, why risk it? I've been dealing with conflicts for the last week or two and would like to avoid more. Thanks

**Comment by Youssef1313** [comment]
@jcouv Would it help if I created two PRs for both `main` and `record-structs` branches with the same commit?

**Comment by jcouv** [comment]
We try limiting duplicating changes. Is this important to fix in 16.9, or is this planned for 17.0? If the latter, then bundle with record-structs.
By the way, the record-structs branch added `DeclarationKind.RecordStruct`, so if we want to deprecate `DeclarationKind.RecordClass`... That is another reason to make the change in the record-structs (or soon `features/compiler`) branch.

**Comment by sharwell** [comment]
This is a workspaces API and myself and @CyrusNajmabadi are in favor of #52803, and @tmat doesn't object to the change. I think we are ready to proceed and will make the team is aware of the pending change.

**Comment by sharwell** [comment]
@jcouv Note that a conflict is unavoidable. We are updating main to correct the use of `RecordClass`, but we cannot automatically update `RecordStruct`. This field will need to be updated prior to or during the merge to the feature branch.

**Comment by jcouv** [comment]
@sharwell The conflict is avoidable if the present change can be made in 17.0 instead of 16.9.

**Comment by Youssef1313** [comment]
@jcouv PR https://github.com/dotnet/roslyn/pull/52803 was already merged. But it may not cause conflicts.
To remove RecordStruct I need `main` to be merged into `features/compiler`, is this something that will be done anytime soon?

**Comment by jcouv** [comment]
@Youssef1313 I'll refresh the `features/compiler` branch today.

**Comment by jcouv** [comment]
@Youssef1313 Here's the PR to refresh `features/compiler` branch: https://github.com/dotnet/roslyn/pull/52973

**Comment by Youssef1313** [comment]
Thanks @jcouv. Opened https://github.com/dotnet/roslyn/pull/52988 to address this issue.

**Comment by Youssef1313** [comment]
Fixed in #52988
